sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -352,6 +351,36 @@ case class Join(
   }
 }
 
+/**
+ * Append data to an existing table.
+ */
+case class AppendData(
+    table: NamedRelation,
+    query: LogicalPlan,
+    isByName: Boolean) extends LogicalPlan {
+  override def children: Seq[LogicalPlan] = Seq(query)
```

Comment:
why is `table` not a child? Then we can't transform the table relation.

Reply:
What transforms would be enabled by making `table` a child? Do we want to transform the relation? It is fixed so I didn't think that was a good idea.

Reply:
I also see that `InsertIntoTable` doesn't list the relation as a child.